docs/development/design-risks-and-mitigations.md

```diff
@@ -0,0 +1,161 @@
+Design Risks and Mitigations
+============================
```

Comment:
I'm not sure where this document belongs. We could move it to the design folder and link to it from the [main design document](https://github.com/gchq/sleeper/blob/develop/docs/design.md). We could also link to it from the [common problems document](https://github.com/gchq/sleeper/blob/develop/docs/common-problems-and-their-solutions.md). It needs a link from somewhere.

Reply:
I've given it it's own link from the README.